streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -811,7 +810,7 @@ public void run() {
      * This will block until all threads have stopped.
      */
     public void close() {
-        close(DEFAULT_CLOSE_TIMEOUT, TimeUnit.SECONDS);
+        close(Long.MAX_VALUE);
```

Comment:
We translate old default of zero to Long.MAX_VALUE within deprecated `close(final long timeout, final TimeUnit timeUnit)` -- we can call `private close()` directly instead.